matrix_sdk/src/room/invited.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{room::Common, Client, Result, Room, RoomType};
+use std::ops::Deref;
+
+/// A room in the invited state.
+///
+/// This struct contains all methodes specific to a `Room` with type `RoomType::Invited`.
+/// Operations may fail once the underlaying `Room` changes `RoomType`.
+#[derive(Debug, Clone)]
+pub struct Invited {
+    inner: Common,
+}
+
+impl Invited {
+    /// Create a new `room::Invited` if the underlaying `Room` has type `RoomType::Invited`.
+    ///
+    /// # Arguments
+    /// * `client` - The client used to make requests.
+    ///
+    /// * `room` - The underlaying room.
+    pub fn new(client: Client, room: Room) -> Option<Self> {
+        if room.room_type() == RoomType::Invited {
+            Some(Self {
+                inner: Common::new(client, room),
+            })
+        } else {
+            None
+        }
+    }
+
+    /// Leave this room.
+    ///
+    /// This essentially ignores an invite.
+    pub async fn leave(&self) -> Result<()> {
+        self.inner.leave().await
+    }
+
+    /// Join this room.
+    pub async fn join(&self) -> Result<()> {
```

Comment:
Not sure if we'll want to rename this as well to `accept_invitation()`, I'm 50/50 about this one.